drivers/overlay/encryption.go

```diff
@@ -219,6 +224,34 @@ func programMangle(vni uint32, add bool) (err error) {
 	return
 }
 
+func programInput(vni uint32, add bool) (err error) {
+	var (
+		p      = strconv.FormatUint(uint64(vxlanPort), 10)
+		c      = fmt.Sprintf("0>>22&0x3C@12&0xFFFFFF00=%d", int(vni)<<8)
+		chain  = "INPUT"
+		rule   = []string{"-p", "udp", "--dport", p, "-m", "u32", "--u32", c, "-j"}
+		accept = append(append([]string{"-m", "policy", "--dir", "in", "--pol", "ipsec"}, rule...), "ACCEPT")
+		block  = append(rule, "DROP")
+		action = iptables.Append
```

Comment:
Change LGTM. But for better readability I would prefer naming the matches explicitly..

```
vni_check =  []string{"-p", "udp", "--dport", p, "-m", "u32", "--u32", c, "-j"}
ipsec_check = append([]string{"-m", "policy", "--dir", "in", "--pol", "ipsec"}, rule...)
accept = append(ipsec_check, "ACCEPT")
block = append(vni_check, "DROP")
```